shell/platform/android/platform_view_android.cc

```diff
@@ -319,6 +319,15 @@ void PlatformViewAndroid::SetSemanticsEnabled(JNIEnv* env,
 
 void PlatformViewAndroid::ReleaseSurface() {
   NotifyDestroyed();
+
+  ftl::AutoResetWaitableEvent latch;
+  blink::Threads::Gpu()->PostTask([this, &latch]() {
+    if (surface_gl_->IsValid())
+      surface_gl_->GLContextClearCurrent();
+    latch.Signal();
+  });
+  latch.Wait();
+
   surface_gl_->TeardownOnScreenContext();
```

Comment:
The `TeardownOnScreenContext()` can also be moved into the task right? Also, I am not sure the latch is needed then.